dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -0,0 +1,254 @@
+    refute_nil user.secret_words
+
+    purge_user user
+
+    user.reload
+    assert_nil user.encrypted_password
+    assert_nil user.reset_password_token
+    assert_nil user.secret_picture
+    assert_nil user.secret_words
+  end
+
+  test 'clears provider uid but not provider type' do
+    user = create :student,
+      provider: 'clever',
+      uid: 'fake-clever-uid'
+    assert_equal user.provider, 'clever'
+    refute_nil user.uid
+
+    purge_user user
+
+    user.reload
+    assert_equal user.provider, 'clever'
+    assert_nil user.uid
+  end
+
+  test 'clears school information' do
+    user = create :teacher,
+      full_address: 'fake-full-address',
+      school: 'fake-school-info',
+      school_info: create(:school_info)
+    refute_nil user.full_address
+    refute_nil user.school
+    refute_nil user.school_info_id
+
+    purge_user user
+
+    user.reload
+    assert_nil user.full_address
+    assert_nil user.school
+    assert_nil user.school_info_id
+  end
+
+  test 'clears user properties' do
+    user = create :teacher,
+      ops_first_name: 'test-value',
+      ops_last_name: 'test-value',
+      district_id: 'test-value',
+      ops_school: 'test-value',
+      ops_gender: 'test-value',
+      using_text_mode: 'test-value',
+      last_seen_school_info_interstitial: 'test-value',
+      ui_tip_dismissed_homepage_header: 'test-value',
+      ui_tip_dismissed_teacher_courses: 'test-value',
+      oauth_refresh_token: 'test-value',
+      oauth_token: 'test-value',
+      oauth_token_expiration: 'test-value',
+      sharing_disabled: 'test-value',
+      next_census_display: 'test-value'
+    refute_equal({}, user.properties)
+
+    purge_user user
+
+    user.reload
+    assert_equal({}, user.properties)
+  end
+
+  test 'leaves urm and races for US users' do
+    user = create :student, :within_united_states, races: 'white,hispanic'
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+
+    purge_user user
+
+    user.reload
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+  end
+
+  test 'clears urm and races for non-US users' do
+    user = create :student, :outside_united_states, races: 'white,hispanic'
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+
+    purge_user user
+
+    user.reload
+    assert_nil user.urm
+    assert_nil user.races
+  end
+
+  test 'removes StudioPerson if it was only tied to this user' do
+    user = create :teacher
+    studio_person = user.studio_person
+    refute_nil user.studio_person_id
+
+    purge_user user
+
+    user.reload
+    assert_nil user.studio_person_id
+    assert_empty StudioPerson.where(id: studio_person.id)
+  end
+
+  test "only removes association with StudioPerson if it's tied to other users" do
+    user = create :teacher
+    studio_person = user.studio_person
+    refute_nil user.studio_person_id
+    other_user = create :teacher, studio_person_id: studio_person.id
```

Comment:
note: since `studio_person` is an association on this id, you can also set it directly:
```ruby
create :teacher, studio_person: studio_person
```